Update simple form components to new theme system

lederer opened the pull request:
## Target Branch
Note this is a request to merge to `feature/theme`, not `develop`. All theme-related PRs should target `feature/theme` until the theme upgrade is complete.

## Overview

Updates the following components to the new theme system:

- Label
- TextInput
- ColorInput
- FileInput
- TextArea
- Checkbox
- Radio
- Select

### Checklist

- [x] Relevant documentation pages have been created or updated
- [ ] Description of PR is in an appropriate section of the changelog and grouped with similar changes if possible
- [ ] Relevant status has been updated on the status page

### Upgrade instructions

If there are any of the following in this PR, provide proper instructions on how to upgrade:

- [x] Changes to the prop names or types of existing components
- [ ] Changes in intended behavior of existing component props
- [ ] Changes in the theme file's structure
- [ ] A required version bump to a non-dev dependency of the project

`size` prop removed from `Label` and `TextInput`

### Notes

This PR does **not** update `Field` and `ToggleField`.

## Testing Instructions

- Use docz pages and/or testbed app to confirm components work as expected

Closes #162 
Closes #165 
Closes #159 
Closes #161 
Closes #166 
Closes #158 
Closes #163 
Closes #164

lederer committed:
Update Label to new theme system

- Remove `size` prop

Closes #162

lederer committed:
Update TextInput to new theme system

- Remove `size` prop

Closes #165

lederer committed:
Update ColorInput to new theme system

Closes #159

lederer committed:
Update FileInput to new theme system

Closes #161

lederer committed:
Update Textarea to new theme system

Closes #166

lederer committed:
Rename Textarea to TextArea

lederer committed:
Update Checkbox to new theme system

Closes #158

lederer committed:
Update Radio to new theme system

Closed #163

lederer committed:
Update Select to new theme system

Close #164

designmatty commented on packages/core/components/checkbox/index.js:
```diff
@@ -101,20 +122,20 @@ const Checkbox = ({
       />
-      <StyledIcon
-        className="toggle--checked"
+      <Icon
```
Do you think we should be applying `aria-hidden="true"` to the `<Icons>` in `<Checkbox>` and `<Radio`>

I was reading this and wanted your thoughts. https://css-tricks.com/accessible-svgs/#article-header-id-10

lederer replied:
Good catch. Yes, you're right.

designmatty commented on packages/core/components/colorInput/index.js:
```diff
@@ -1,38 +1,32 @@
-  }}
+  padding: ${themeGet("colorInput.space.p")};
+  border: ${themeGet("colorInput.borders.border")};
+  borderRadius: ${themeGet("colorInput.radii.borderRadius")};
```
Invalid css attritbute here. Also worth noting, border and border-radius will do nothing here. If we intend on allowing these types of overrides we'll have to target the Pseudo-elements with the shadow dom.

See: https://gist.github.com/webtobesocial/aefd6e25064c08e0cc9a#input-typecolor for available pseudo-elements

lederer replied:
Thanks for catching the typo. 

`border` and `border-radius` do affect this component for me on Chrome.

designmatty replied:
oh my mistake, you're right.

lederer committed:
Add aria-hidden to decorative SVGs

designmatty commented on packages/core/components/fileInput/index.js:
```diff
@@ -1,23 +1,32 @@
+
+  padding: ${themeGet("fileInput.space.p")};
+  border: ${themeGet("fileInput.borders.border")};
+  borderRadius: ${themeGet("fileInput.radii.borderRadius")};
```
Invalid CSS attribute. 
We might also consider adding the `BACKGROUND` constant to support these types of styles:
![image](https://user-images.githubusercontent.com/1928955/61814719-a6a73400-ae16-11e9-8ea8-78e894bfad98.png)

lederer replied:
Will do.

designmatty commented on packages/core/components/radio/index.js:
```diff
@@ -135,10 +154,7 @@ Radio.defaultProps = {
+  uncheckedIcon: 'radioUnchecked'
 };
 
 export default Radio;
```
There seem to be a handful of issues with `Radio`

In the attached screenshot I added a border to the `span` surrounding which reveals some strange line-height and alignment issues. The border represent clickable areas as well. For some reason, the shape/size and position of the border changes drastically when you change the `size` prop. 

I also noticed that props `uncheckedColor` and `checkedColor` don't work.

![image](https://user-images.githubusercontent.com/1928955/61815690-d0f9f100-ae18-11e9-88c8-17d9d19d62ec.png)

designmatty replied:
It seems that `<Checkbox>` has all of the same issues.

![image](https://user-images.githubusercontent.com/1928955/61816247-f89d8900-ae19-11e9-8845-d7ef443950fd.png)

designmatty replied:
We might need to consider a different approach where we don't use icons, but instead, we might use pure CSS. Along with some of the odd behavior above, there may be more unknown issues presented when a user changes those icons in the theme. This is obviously more limiting in how we can customize, but it may prove to be more stable components.

lederer replied:
I pushed fixes for the color props and size prop issues.

But I can't recreate the click area issue, captured in the borders you added. When I recreate (in Chrome) the conditions you had, the click areas match the visible areas. 

Do the fixes I just pushed fix that for you?

designmatty replied:
Those changes fixed the issue for me.

lederer committed:
Fixes

- Fix typos
- Add bg prop to fileInput
- Fix Radio and Checkbox sizing issues
- Fix Radio and Checkbox color props

lederer committed:
Fixup

lederer committed:
Add color to FileInput theme

designmatty reviewed (approved):
Thanks for fixing those issues. I think this should be good to merge in
